api/v1/verticadb_webhook.go

```diff
@@ -1882,33 +1874,22 @@ func (v *VerticaDB) checkShutdownSandboxImage(old runtime.Object, allErrs field.
 	return allErrs
 }
 
-// checkTerminatingSandboxes ensure the sandbox to terminate is not shut down
-func (v *VerticaDB) checkTerminatingSandboxes(old runtime.Object, allErrs field.ErrorList) field.ErrorList {
-	oldObj := old.(*VerticaDB)
+// checkShutdownForSandboxesToBeRemoved ensures the sandbox to terminate is not shut down
+func (v *VerticaDB) checkShutdownForSandboxesToBeRemoved(oldObj *VerticaDB, allErrs field.ErrorList) field.ErrorList {
 	oldSandboxMap := oldObj.GenSandboxMap()
 	newSandboxMap := v.GenSandboxMap()
 	sandboxesToBeRemoved := vutil.MapKeyDiff(oldSandboxMap, newSandboxMap)
-	newSubclusterMap := v.GenSubclusterMap()
 	for _, sandboxToBeRemoved := range sandboxesToBeRemoved {
 		if !oldSandboxMap[sandboxToBeRemoved].Shutdown {
 			continue
-		}
-		numOfSclustersInSbox := len(oldSandboxMap[sandboxToBeRemoved].Subclusters)
-		sclustersToTerminate := numOfSclustersInSbox
-		for _, subcluster := range oldSandboxMap[sandboxToBeRemoved].Subclusters {
-			if _, ok := newSubclusterMap[subcluster.Name]; ok {
-				sclustersToTerminate--
-			}
-		}
-		// sandboxToBeRemoved and all its subclusters are not found in new spec
-		if sclustersToTerminate == numOfSclustersInSbox {
+		} else {
```

Comment:
You can remove the else here since you already used a "continue" in line 1884

Reply:
with the "else", it is easier to read and understand the logic.

Reply:
Then line 1884 doesn't make sense. You can just use `if oldSandboxMap[sandboxToBeRemoved].Shutdown`, what's the point to use an else block.